### ISSUE TITLE
uninitialized read (default "last request time" not set)

### ISSUE THREAD

**Issue by whtghst1** [Haivision/srt#343]
This was also in the core UDT 4.11 library.

It can be fixed with the following in the constructor:

```
    m_bBroken = false;
    m_bPeerHealth = true;
    m_ullLingerExpiration = 0;
+   m_llLastReqTime = 0;
 }
```

**Comment by ethouris** [comment]
This value is preset in both `startConnect` and `processAsyncConnectRequest` (and not to 0, but to current time), that's why probably it hasn't led so far to any problems. Still, leaving a singular value in the variable may always lead to problems - thanks for having found it.

**Comment by whtghst1** [comment]
here is a patch for the current master for this.

```
@@ -179,6 +179,7 @@ void CUDT::construct()
     m_bBroken = false;
     m_bPeerHealth = true;
     m_ullLingerExpiration = 0;
+	m_llLastReqTime = 0;
 
     m_lSrtVersion = SRT_DEF_VERSION;
     m_lPeerSrtVersion = 0; // not defined until connected.
```

**Comment by maxlovic** [comment]
Fixed by PR #350
